src/main/java/mate/jdbc/dao/impl/CarDaoImpl.java

```diff
@@ -0,0 +1,249 @@
+            return statement.executeUpdate() > 0;
+        } catch (SQLException e) {
+            throw new DataProcessingException("Couldn't delete car with id " + id, e);
+        }
+    }
+
+    @Override
+    public List<Car> getAllByDriver(Long driverId) {
+        List<Long> carsId = new ArrayList<>();
+        String getAllByDriverRequest =
+                "SELECT cars_id\n"
+                + "FROM cars_drivers cd\n"
+                + "JOIN cars c ON cd.cars_id = c.id\n"
+                + "WHERE drivers_id = ? AND c.is_deleted = FALSE;";
+        try (Connection connection = ConnectionUtil.getConnection();
+                 PreparedStatement statement =
+                         connection.prepareStatement(getAllByDriverRequest)) {
+            statement.setLong(FIRST_INDEX, driverId);
+            ResultSet resultSet = statement.executeQuery();
+            while (resultSet.next()) {
+                carsId.add(resultSet.getLong(FIRST_INDEX));
+            }
+        } catch (SQLException e) {
+            throw new DataProcessingException("Can't get all cars from DB", e);
+        }
+        List<Car> cars = new ArrayList<>();
+        extractCarsByCarsId(carsId, cars);
+        return cars;
+    }
+
+    private void extractCarsByCarsId(List<Long> carsId, List<Car> cars) {
+        if (!carsId.isEmpty()) {
+            for (Long carId : carsId) {
+                cars.add(get(carId).orElseThrow(() ->
+                        new DataProcessingException("Can't get car by id: "
+                                                    + carId + " from cars DB.")));
+            }
+        }
+    }
+
+    private void deletePreviousRelations(Long id) {
+        Car previousCarVersion = this.get(id).orElseThrow(() ->
+                new DataProcessingException("Can't get existing variant of car with id: "
+                                            + id + " from cars DB."));
+        if (previousCarVersion.getDrivers().isEmpty()) {
+            return;
+        }
+        String insertRequest = "DELETE FROM cars_drivers\n "
+                               + "WHERE cars_id = ? AND drivers_id = ?;";
+        try (Connection connection = ConnectionUtil.getConnection();
+                 PreparedStatement statement = connection.prepareStatement(insertRequest,
+                         Statement.RETURN_GENERATED_KEYS)) {
+            statement.setLong(FIRST_INDEX, previousCarVersion.getId());
+            for (Driver driver : previousCarVersion.getDrivers()) {
+                statement.setLong(SECOND_INDEX, driver.getId());
+                statement.executeUpdate();
+            }
+        } catch (SQLException e) {
+            throw new DataProcessingException(
+                    "Couldn't delete "
+                    + "relations between " + previousCarVersion
+                    + " and drivers.", e);
+        }
+    }
+
+    private Car addDriverToCar(ResultSet resultSet, Car foundedCar) throws SQLException {
```

Comment:
It is better to change the signature: 
```suggestion
    private Car addDriverToCar(Car car, Driver driver) throws SQLException {
```
addDriverToCar(foundedCar, initializeDriver(resultSet));

Such style makes your methods more decoupled and flexible.